### DISCUSSION TITLE
Implement custom flow without dashboard with raw c# code

### DISCUSSION THREAD

**Discussion by abhalesd** [elsa-workflows/elsa-core#5642] [Design & tech]
Hi,
I want to implement custom flow where activities will be my custom functions written in c#. I was able to do that, But I am not able to write a code for connections to decide which one should be executed. Do you have any example of flowchart completely implemented through c# code using connections, activities.

Below is the example which I want to implement in elsa workflow 

![image](https://github.com/elsa-workflows/elsa-core/assets/41250078/f10622b6-0684-44d6-b8bf-50ffb6d71b05)

**Comment by glime-ai** [comment]
To implement a flowchart with custom activities and connections in Elsa Workflow using C#, you can follow the example provided in the context from the source "https://v3.elsaworkflows.io/docs/core-concepts/programmatic-vs-designer". This example demonstrates how to define a workflow variable, create activities, and set up connections between them programmatically.

Here's a simplified version of how you might define a workflow similar to the one in the image you provided:

```csharp
// Define workflow variables if needed
var someVariable = new Variable<string>();

// Define your custom activities
var activity1 = new YourCustomActivity1();
var activity2 = new YourCustomActivity2();
var activity3 = new YourCustomActivity3();
// ... define other activities as needed

// Define a flowchart workflow
var workflow = new Flowchart
{
    // Register variables if you have any
    Variables = { someVariable },
    
    // Add the activities
    Activities =
    {
        activity1,
        activity2,
        activity3,
        // ... add other activities
    },
    
    // Setup the connections between activities
    Connections =
    {
        new Connection(activity1, activity2, outcome: "Outcome1"),
        new Connection(activity1, activity3, outcome: "Outcome2"),
        // ... add other connections based on your custom outcomes
    }
};

// To run the workflow, you would typically use a workflow host or runner
var workflowHost = ...; // Obtain a reference to your workflow host
await workflowHost.StartWorkflowAsync(workflow);
```

In this code, `YourCustomActivity1`, `YourCustomActivity2`, and `YourCustomActivity3` represent your custom activities. You would replace these with your actual activity classes. The `Connection` objects define how the activities are linked together, with the `outcome` parameter specifying which outcome of the source activity should trigger the transition to the target activity.

Please note that the actual implementation details may vary depending on the version of Elsa Workflow you are using and the specifics of your custom activities. You may need to refer to the documentation for the exact syntax and available methods.

For more detailed examples and guidance, you can refer to the Elsa documentation and the provided sources, especially the "Programmatic workflows vs designer workflows" section, which outlines how to programmatically create workflows in Elsa.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by kyle-alm** [reply]
In the example in the documentation here:  "https://v3.elsaworkflows.io/docs/core-concepts/programmatic-vs-designer", the workflow is not dependent on outcomes and does not branch and is effectively just a Sequence. I would think anyone wanting to define a Flowchart would be wanting to do so based on outcomes. The code you provided above appears to do that, but it's incorrect, as Connection has no constructor that takes three arguments. How can we make a connection based on outcomes?
